bids-validator/src/types/columns.test.ts

```diff
@@ -0,0 +1,19 @@
+import { assertEquals } from '../deps/asserts.ts'
+import { ColumnsMap } from './columns.ts'
+
+Deno.test('ColumnsMap', async (t) => {
+  await t.step('preserves insertion order with property access', () => {
+    const columns = new ColumnsMap()
+    columns['a'] = ['0']
+    columns[2] = ['1']
+    columns[1] = ['2']
+    columns['b'] = ['3']
+
+    // Check that the order is still correct with iteration
+    let iteration = 0
+    for (const [_, val] of columns) {
+      assertEquals(val, [iteration.toString()])
+      iteration += 1
+    }
```

Comment:
Do we also have property access, or is it just square-bracket access `[]`?

```suggestion
    }

    // Check property access
    assertEquals(columns.a, '0')
```

Is there a way to get property access, if not?

Reply:
Square bracket and dot access to properties is equivalent in JavaScript but it did point out a bug with how the name is looked up. aba7edc includes a fix and your suggested test change.